envelope.go

```diff
@@ -209,6 +211,7 @@ func (w *envelopeWriter) marshal(message any) *Error {
 func (w *envelopeWriter) write(env *envelope) *Error {
 	if _, err := w.sender.Send(env); err != nil {
 		err = wrapIfContextError(err)
+		err = wrapWithContextError(w.ctx, err)
```

Comment:
Perhaps this would be better to push the error classification into `sender.Send`? The sender (the `duplexHTTPCall`) has a reference to the request, which should also include the context.

Also, under what conditions would we ever call only one or the other of those context-wrap functions? Feels like they want to be a single function -- maybe even a method on `duplexHTTPCall` (so it can pull out the request context for the comparisons in the latter functions).

Reply:
We call wrapWithContext more liberally to convert context errors but not to wrap errors with context codes. We will always call `wrapIfContext` with `wrapWithContext` but still need `wrapIfContext` separate so left them split out. A method on `duplexHTTPCall` would need a corresponding method on the handler side. The marshallers are used on both client/server so can handle it uniformly there.

Reply:
> We will always call `wrapIfContext` with `wrapWithContext` but still need `wrapIfContext` separate so left them split out.

Why do they need to be split? Under what cases do we care about the error _being_ a context error but if the context is actually done (or vice versa)? Also, the names are rather confusing. Just reading the above sentence is really not clear as to which is which.

Reply:
We check for context errors on non IO related errors like the wrapper to the stream handlers. I've merged the call to the `wrapIfContextError` from `wrapIfContextDone`.

Reply:
https://github.com/emcfarlane/connect-go/blob/c3324072f50f3882a98e4ee846ee7511d3911efa/error.go#L279

Which we don't currently have an easy way to access the context from.